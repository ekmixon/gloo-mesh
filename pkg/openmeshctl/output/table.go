package output

// Table is used to represent an object in a tabular manner.
type Table struct {
	Headers []string
	NextRow func() []string

	nextRowData []string
}

// HasNextRow advances the table to the next row.
func (t *Table) HasNextRow() bool {
	row := t.NextRow()
	if row == nil {
		return false
	}

	t.nextRowData = row
	return true
}

// GetNextRow returns the current data row.
func (t *Table) GetNextRow() []string {
	return t.nextRowData
}
