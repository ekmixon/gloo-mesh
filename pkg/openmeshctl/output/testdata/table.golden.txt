NAMESPACE   NAME                                               MESH
gloo-mesh   details-bookinfo-mgmt-cluster                      gloo-mesh.istiod-istio-system-mgmt-cluster
gloo-mesh   details-bookinfo-remote-cluster                    gloo-mesh.istiod-istio-system-remote-cluster
gloo-mesh   istio-ingressgateway-istio-system-mgmt-cluster     gloo-mesh.istiod-istio-system-mgmt-cluster
gloo-mesh   istio-ingressgateway-istio-system-remote-cluster   gloo-mesh.istiod-istio-system-remote-cluster
gloo-mesh   productpage-bookinfo-mgmt-cluster                  gloo-mesh.istiod-istio-system-mgmt-cluster
gloo-mesh   productpage-bookinfo-remote-cluster                gloo-mesh.istiod-istio-system-remote-cluster
gloo-mesh   ratings-bookinfo-mgmt-cluster                      gloo-mesh.istiod-istio-system-mgmt-cluster
gloo-mesh   ratings-bookinfo-remote-cluster                    gloo-mesh.istiod-istio-system-remote-cluster
gloo-mesh   reviews-bookinfo-mgmt-cluster                      gloo-mesh.istiod-istio-system-mgmt-cluster
gloo-mesh   reviews-bookinfo-remote-cluster                    gloo-mesh.istiod-istio-system-remote-cluster
